src/jit/valuenum.cpp

```diff
@@ -1613,6 +1613,7 @@ double ValueNumStore::GetConstantDouble(ValueNum argVN)
 {
     assert(IsVNConstant(argVN));
     var_types argVNtyp = TypeOfVN(argVN);
+    assert(varTypeIsFloating(argVNtyp));
```

Comment:
Yeah... in fact, it looks like we're already (without this change) constant-folding the single-precision intrinsics, just doing it by widening to double and evaluating and truncating back down, is that right?  The language in the [spec](https://www.ecma-international.org/publications/standards/Ecma-335.htm) certainly seems to give us the liberty to do that (I.12.1.3). I'm inclined to say that we should either go ahead and make this change for all constant-folding on fp ops or not make this change, to keep things consistent.

I'm not sure whether it's best to continue folding at double precision or to switch to single.  I'd think we'd want to minimize differences between optimized and unoptimized code's results, and I think the unoptimized code would end up happening at whatever precision our back-end uses for singles, which maybe has a greater chance of matching the C++ compiler's handling of `float` than of `double`, which would argue for continuing this change... but I'm no floating-point expert.  Maybe someone else wants to chime in. @sivarv? @CarolEidt?